pkg/resource/parameter_group/post_set_output.go

```diff
@@ -0,0 +1,90 @@
+package parameter_group
+
+import (
+	"context"
+	svcapitypes "github.com/aws-controllers-k8s/memorydb-controller/apis/v1alpha1"
+	ackerr "github.com/aws-controllers-k8s/runtime/pkg/errors"
+	svcsdk "github.com/aws/aws-sdk-go/service/memorydb"
+)
+
+func (rm *resourceManager) setParameters(
+	ctx context.Context,
+	ko *svcapitypes.ParameterGroup,
+) (*svcapitypes.ParameterGroup, error) {
+	if ko.Spec.ParameterGroupName == nil {
+		return ko, nil
+	}
+
+	parameters, err := rm.describeParameters(ctx, ko.Spec.ParameterGroupName)
+	if err != nil {
+		return nil, err
+	}
+	parameterNameValues := []*svcapitypes.ParameterNameValue{}
+	for _, p := range parameters {
+		sp := svcapitypes.ParameterNameValue{
+			ParameterName:  p.Name,
+			ParameterValue: p.Value,
+		}
+		parameterNameValues = append(parameterNameValues, &sp)
+	}
+	ko.Spec.ParameterNameValues = parameterNameValues
+	// Update parameter value to nil to not duplicate status and spec values.
+	for _, p := range parameters {
+		p.Value = nil
+	}
+	ko.Status.Parameters = parameters
+	return ko, nil
+}
+
+func (rm *resourceManager) describeParameters(
+	ctx context.Context,
+	parameterGroupName *string,
+) ([]*svcapitypes.Parameter, error) {
+	parameters := []*svcapitypes.Parameter{}
+
+	input, err := rm.newDescribeParametersRequestPayload(parameterGroupName)
+	if err != nil {
+		return nil, err
+	}
+	response, respErr := rm.sdkapi.DescribeParametersWithContext(ctx, input)
```

Comment:
It appears this method returns a [`NextToken`](https://docs.aws.amazon.com/sdk-for-go/api/service/memorydb/#DescribeParametersOutput) used for pagination. Check out how we handled that in the [IAM controller](https://github.com/aws-controllers-k8s/iam-controller/blob/main/pkg/resource/role/hooks.go#L217-L230).

Reply:
API would return all the values and doesn't need pagination.

Reply:
> API would return all the values and doesn't need pagination.

But as I linked, the `DescribeParametersOutput` contains a `NextToken` field, with the comment:

> An optional argument to pass in case the total number of records exceeds
> the value of MaxResults. If nextToken is returned, there are more results
> available. The value of nextToken is a unique pagination token for each page.
> Make the call again using the returned token to retrieve the next page. Keep
> all other arguments unchanged.

Which implies the result is paginated and needs to be called multiple times to retrieve all the values?